server/src/internalClusterTest/java/org/opensearch/index/codec/MultiCodecMergeIT.java

```diff
@@ -0,0 +1,175 @@
+            forceMergeMultipleCodecs(codec.getKey(), codec.getValue(), codecMap);
+        }
+
+    }
+
+    private void forceMergeMultipleCodecs(String finalCodec, String finalCodecMode, Map<String, String> codecMap) throws ExecutionException,
+        InterruptedException {
+
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        final String index = "test-index" + finalCodec;
+
+        // creating index
+        createIndex(
+            index,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+                .put("index.codec", "default")
+                .put("index.merge.policy.max_merged_segment", "1b")
+                .build()
+        );
+        ensureGreen(index);
+        // ingesting and asserting segment codec mode for all four codecs
+        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+            useCodec(index, codec.getKey());
+            ingestDocs(index);
+        }
+
+        assertTrue(
+            getSegments(index).stream()
+                .flatMap(s -> s.getAttributes().values().stream())
+                .collect(Collectors.toSet())
+                .containsAll(codecMap.values())
+        );
+
+        // force merge into final codec
+        useCodec(index, finalCodec);
+        flushAndRefreshIndex(index);
+        final ForceMergeResponse forceMergeResponse = client().admin().indices().prepareForceMerge(index).setMaxNumSegments(1).get();
+
+        assertThat(forceMergeResponse.getFailedShards(), is(0));
+        assertThat(forceMergeResponse.getSuccessfulShards(), is(2));
+
+        flushAndRefreshIndex(index);
+
+        List<Segment> segments = getSegments(index).stream().filter(Segment::isSearch).collect(Collectors.toList());
+        assertEquals(1, segments.size());
+        assertTrue(segments.stream().findFirst().get().attributes.containsValue(finalCodecMode));
+    }
+
+    private void useCodec(String index, String codec) throws ExecutionException, InterruptedException {
+        assertAcked(client().admin().indices().prepareClose(index));
+
+        assertAcked(
+            client().admin()
+                .indices()
+                .updateSettings(new UpdateSettingsRequest(index).settings(Settings.builder().put("index.codec", codec)))
+                .get()
+        );
+
+        assertAcked(client().admin().indices().prepareOpen(index));
+    }
+
+    private void ingestDocs(String index) throws InterruptedException {
+        ingest(index);
+        flushAndRefreshIndex(index);
+    }
+
+    private ArrayList<Segment> getSegments(String index) {
+
+        return new ArrayList<>(
+            client().admin()
+                .indices()
+                .segments(new IndicesSegmentsRequest(index))
+                .actionGet()
+                .getIndices()
+                .get(index)
+                .getShards()
+                .get(0)
```

Comment:
Though the index contains 2 shards, only the segments of first shard is fetched here. We can either make change it to 1 shard or fetch segments of both the shards here.